src/vm/memory/segments.zig

```diff
@@ -119,19 +121,55 @@ test "memory segment manager" {
     const relocatable_address_1 = memory_segment_manager.addSegment();
 
     // Check that the memory segment manager has one segment.
-    try std.testing.expect(memory_segment_manager.memory.num_segments == 1);
+    try expect(memory_segment_manager.memory.num_segments == 1);
 
     // Check if the relocatable address is correct.
-    try std.testing.expect(relocatable_address_1.segment_index == 0);
-    try std.testing.expect(relocatable_address_1.offset == 0);
+    try expect(relocatable_address_1.segment_index == 0);
+    try expect(relocatable_address_1.offset == 0);
```

Comment:
Here as we are testing the 2 elements of the structure which only has 2, would it not be better to make an expectation of equality on the complete structure. In this way, if we have to add fields to the structure one day, we will be forced to modify the test otherwise it will no longer pass and this completely secures the testing. To me it would look something like:

```ziglang
try std.testing.expectEqual(
  relocatable.Relocatable{
      .segment_index = 0,
      .offset = 0,
  },
relocatable_address_1,
)
```